namerd/iface/grpc/src/main/scala/io/buoyant/namerd/iface/grpc/InterpreterClient.scala

```diff
@@ -0,0 +1,174 @@
+package io.buoyant.namerd
+package iface.grpc
+
+import com.twitter.concurrent.AsyncQueue
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.h2
+import com.twitter.finagle.naming.NameInterpreter
+import com.twitter.finagle.stats.StatsReceiver
+import com.twitter.io.Buf
+import com.twitter.util.{Activity, Closable, Future, Return, Throw, Try, Var}
+import io.buoyant.grpc.runtime.{GrpcStatus, Stream}
+import io.buoyant.namer.{ConfiguredDtabNamer, Delegator, DelegateTree}
+import io.buoyant.proto.{BoundNameTree, BoundDelegateTree}
+import io.buoyant.proto.namerd.{Interpreter => ProtoInterpreter, Addr => _, _}
+import scala.util.control.NoStackTrace
+
+object InterpreterClient {
+  import InterpreterProto._
+
+  def apply(client: ProtoInterpreter, namespace: String): NameInterpreter =
+    Impl(client, namespace)
+
+  case class Impl(client: ProtoInterpreter, namespace: String)
+    extends NameInterpreter with Delegator {
+
+    override def bind(dtab: Dtab, path: Path): Activity[NameTree[Name.Bound]] = {
+      val req = mkBindReq(namespace, path, dtab)
+      val run = Var.async[Activity.State[NameTree[Name.Bound]]](Activity.Pending) { state =>
+        val trees = client.streamBoundTree(req)
+
+        @volatile var closed = false
+        def loop(): Future[Unit] =
+          if (closed) Future.Unit
+          else trees.recv().transform {
+            case Throw(e) =>
```

Comment:
We should retry the request on a stream failure.  Is that what your TODO alludes to?